service/watcher/chartstatus/dynclient.go

```diff
@@ -65,7 +66,13 @@ func (c *ChartStatusWatcher) waitForAvailableConnection(ctx context.Context, dyn
 		// List all chart CRs in the target cluster to confirm the connection
 		// is active and the chart CRD is installed.
 		_, err = dynClient.Resource(chartResource).Namespace(c.chartNamespace).List(ctx, metav1.ListOptions{})
-		if err != nil {
+		if tenant.IsAPINotAvailable(err) {
+			c.logger.Debugf(ctx, "workload cluster is not available")
+			return microerror.Mask(err)
+		} else if IsResourceNotFound(err) {
+			c.logger.Debugf(ctx, "chart CRD is not installed")
+			return microerror.Mask(err)
```

Comment:
The errors can be cryptic. So handling each expected error and logging why the watcher can't proceeed.